flake.nix

```diff
@@ -29,6 +29,17 @@
     eachDefaultSystem (
       system:
       let
+        assertVersion =
+          version: pkg:
+          (
+            assert (
+              pkgs.lib.assertMsg (builtins.toString pkg.version == version) ''
+                Expecting version of ${pkg.name} to be ${version} but got ${pkg.version};
```

Comment:
Umm, if I got this error message I would never think of looking in that `.ruby-version` file!
It would be nice to say where the two different version numbers are coming from so that you can debug it.